BitList.cs

```diff
@@ -0,0 +1,1737 @@
+            ToSByteArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(ushort[] array, int arrayIndex)
+        {
+            ToUInt16Array().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(uint[] array, int arrayIndex)
+        {
+            ToUInt32Array().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(ulong[] array, int arrayIndex)
+        {
+            ToUInt64Array().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(float[] array, int arrayIndex)
+        {
+            ToSingleArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(double[] array, int arrayIndex)
+        {
+            ToDoubleArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(decimal[] array, int arrayIndex)
+        {
+            ToDecimalArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(char[] array, int arrayIndex)
+        {
+            ToCharArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Get a the value in the specified index.
+        /// </summary>
+        /// <exception cref="IndexOutOfRangeException" />
+        public bool Get(int index)
+        {
+            if (!IsEmpty && index < bits.Length && index >= 0)
+            {
+                return bits[index];
+            }
+            else
+            {
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Get a range of value in the specified index.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException" />
+        public BitList Get(int index, int count)
+        {
+            if (!IsEmpty && index < bits.Length && index >= 0 && index + count <= bits.Length - index && count > 0)
```

Comment:
Codacy found an issue: [Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)